cmd/cvp-trigger/main.go

```diff
@@ -4,69 +4,190 @@ import (
+	if !exists {
+		return fmt.Errorf("validating job config path %s failed, does not exist", o.jobConfigPath)
+	}
+
+	if o.jobName == "" {
+		return fmt.Errorf("required parameter %s was not provided", jobNameOption)
+	}
+
+	if o.ocpVersion == "" {
+		return fmt.Errorf("required parameter %s was not provided", ocpVersionOption)
+	}
+	if !strings.HasPrefix(o.ocpVersion, "4") {
+		return fmt.Errorf("ocp-version must be 4.x or higher")
+	}
+
+	if o.packageName == "" {
+		return fmt.Errorf("required parameter %s was not provided", packageNameOption)
+	}
+
+	if o.prowConfigPath == "" {
+		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	}
+	exists, _ = afs.Exists(o.prowConfigPath)
+	if !exists {
+		return fmt.Errorf("validating prow config path %s failed, does not exist", o.prowConfigPath)
+	}
+
+	if !o.dryRun {
+		if o.outputPath == "" {
+			return fmt.Errorf("required parameter %s was not provided", outputFilePathOption)
+		}
+		if !exists {
+			return fmt.Errorf("validating output file path %s failed, does not exist", o.outputPath)
+		}
+	}
+
+	return nil
+}
+
+// getProwJob returns a Prow Job or an error if the provided job name is not found
+func getProwJob(jobName string, config *prowconfig.Config) (*pjapi.ProwJob, error) {
+	var selectedJob *prowconfig.Periodic
+	for _, job := range config.AllPeriodics() {
+		if job.Name == jobName {
+			selectedJob = &job
+			break
+		}
+	}
+
+	if selectedJob == nil {
+		return nil, fmt.Errorf("failed to find the job: %s", jobName)
+	}
+
+	prowjob := pjutil.NewProwJob(pjutil.PeriodicSpec(*selectedJob), nil, nil)
+	return &prowjob, nil
+}
+
+// addEnvVarsToJobSpec adds the given key/value sets as EnvVars in the Env var list of the
+// .Spec.PodSpec.Containers[0].Env
+func addEnvVarsToJobSpec(job *pjapi.ProwJob, envVars map[string]string) {
+	// map key order is random and testing likes to have env vars in predictable order.
+	// create an array, place our map keys in it, sort them, then add them to the envVar
+	// array in that order. #NOTE: ensure that the testing file has our added vars in
+	// sorted order at the end of the env var list.
+	keys := make([]string, 0, len(envVars))
 
-	if o.periodic == "" {
-		return fmt.Errorf("required parameter %s was not provided", periodicOption)
+	for k := range envVars {
+		keys = append(keys, k)
 	}
+	sort.Strings(keys)
 
+	for _, k := range keys {
+		job.Spec.PodSpec.Containers[0].Env = append(job.Spec.PodSpec.Containers[0].Env, v1.EnvVar{Name: k, Value: envVars[k]})
+	}
+}
+
+// writeOutputFile writes the returned JSON to the specified file
+func writeOutputFile(json []byte, path string) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("failed to create output file: %s", path)
+	}
+	defer f.Close()
+	_, err = f.Write(json)
+	_ = f.Sync()
 	return nil
 }
 
 func main() {
-	o := gatherOptions()
-	if err := o.validate(); err != nil {
+	o.gatherOptions()
+	_ = fs.Parse(os.Args[1:])
```

Comment:
Perhaps I'm mistaken, but since we're setting `var fs = flag.NewFlagSet(os.Args[0], flag.ExitOnError)` this automatically calls `os.Exit(2)` on error state. As such, there's no error to handle in this scenario.

Reply:
Sure, but nothing is guaranteeing that someone does not change that call. It's not idiomatic to ignore errors and it's strictly more correct to handle it, so let's try to do that.

Reply:
updated to handle error